src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+  getIsOpen,
+  getTrackId,
+  getTrackTitle
+} from 'audius-client/src/common/store/ui/add-to-playlist/selectors'
+import { toast } from 'audius-client/src/common/store/ui/toast/slice'
+import { newCollectionMetadata } from 'audius-client/src/schemas'
+import { FEED_PAGE, playlistPage } from 'audius-client/src/utils/route'
+import {
+  NativeScrollEvent,
+  NativeSyntheticEvent,
+  ScrollView,
+  StyleSheet,
+  View
+} from 'react-native'
+import { Shadow } from 'react-native-shadow-2'
+
+import Button, { ButtonType } from 'app/components/button'
+import { Card } from 'app/components/card'
+import Drawer from 'app/components/drawer'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { usePushRouteWeb } from 'app/hooks/usePushRouteWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import { ThemeColors } from 'app/utils/theme'
+
+const messages = {
+  title: 'Add To Playlist',
+  addedToast: 'Added To Playlist!',
+  createdToast: 'Playlist Created!'
+}
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    cardContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      justifyContent: 'flex-start',
+      padding: 12,
+      paddingTop: 24
+    },
+    card: {
+      flex: 1,
+      flexBasis: '40%',
+      maxWidth: '50%',
+      marginBottom: 8
+    },
+    newPlaylistButton: {
+      backgroundColor: themeColors.white,
+      height: 48,
+      width: 256
+    },
+    newPlaylistButtonContainer: {
+      backgroundColor: 'transparent'
+    },
+    newPlaylistButtonText: {
+      color: themeColors.neutral
+    }
+  })
+
+export const AddToPlaylistDrawer = () => {
+  const styles = useThemedStyles(createStyles)
+  const dispatchWeb = useDispatchWeb()
+  const pushRouteWeb = usePushRouteWeb()
+  const isOpen = useSelectorWeb(getIsOpen)
+  const trackId = useSelectorWeb(getTrackId)
+  const trackTitle = useSelectorWeb(getTrackTitle)
+  const user = useSelectorWeb(getAccountWithOwnPlaylists)
+  const [isDrawerGestureSupported, setIsDrawerGestureSupported] = useState(true)
+
+  const handleClose = useCallback(() => {
+    dispatchWeb(close())
+  }, [dispatchWeb])
+
+  if (!user || !trackId || !trackTitle) return null
+  const userPlaylists = user.playlists ?? []
+
+  const addToNewPlaylist = () => {
+    const metadata = newCollectionMetadata({
+      playlist_name: trackTitle,
+      is_private: false
+    })
+    const tempId = `${Date.now()}`
+    dispatchWeb(
+      createPlaylist(tempId, metadata, CreatePlaylistSource.FROM_TRACK, trackId)
+    )
+    dispatchWeb(addTrackToPlaylist(trackId!, tempId))
+    toast({ content: messages.createdToast })
+    // TODO: Update the fromPage location. Need to figure out how to get page that the user came from
+    pushRouteWeb(playlistPage(user.handle, trackTitle, tempId), FEED_PAGE)
```

Comment:
also a headsup! we have a new `useNavigation` hook that can handle the pushRouteWeb and the react-navigation useNavigate, which we could potentially use here, there is an example in the `TrackTile` component for `goToArtistPage`

Reply:
Going to make this update in another PR. Need to update the routes in order to go directly to a user playlist page which is not set up just yet